app/src/main/java/com/kks/nimblesurveyjetpackcompose/util/extensions/NetworkExtension.kt

```diff
@@ -72,3 +65,19 @@ inline fun <reified T>Response<*>.parseErrJsonResponse(): T? {
     }
     return null
 }
+
+fun <T> mapError(resourceState: ResourceState<T>, isError: MutableStateFlow<ErrorModel>) {
+    when (resourceState) {
+        is ResourceState.Error ->
+            isError.value =
+                ErrorModel(errorType = ErrorType.INFO, errorMessage = resourceState.error.orEmpty())
+        is ResourceState.NetworkError -> isError.value = ErrorModel(errorType = ErrorType.NETWORK)
+        else -> {
+            // Do nothing
+        }
+    }
+}
```

Comment:
Perhaps we should refactor this to return the `ErrorModel` instead of directly setting it on the `flow`, it will be more reusable that way 🤔 
```suggestion
fun <T> ResourceState<T>.mapError(): ErrorModel? {
    return when (this) {
        is ResourceState.Error -> ErrorModel(errorType = ErrorType.INFO, errorMessage = error.orEmpty())
        is ResourceState.NetworkError -> ErrorModel(errorType = ErrorType.NETWORK)
        else -> null
    }
}
```